### PR TITLE
Refactor Timing & Sensitivity

### LINKED ISSUE #301: Refactor: Timing Sensitivity

### PR DESCRIPTION
**Pull request by stevefosterwta**
Closes github issue 301

# Description of Work
Refactors the TimingSensitivityCollectionViewController

## Notes to Test
The changes in this PR assumes the following feature flag configuration:
- `EditPhrasesEnabled` is not enabled 
- `RefactoredInterfaceEnabled` is enabled 

Screens updated:
Only the Timing and Sensitivity screen was updated in this PR.

### CODE REVIEW

File: Vocable/Features/Settings/TimingSensitivity/TimingSensitivityViewController.swift [hunk 1]
```diff
@@ -8,12 +8,160 @@
...
+        super.traitCollectionDidChange(previousTraitCollection)
+
+        collectionView.setCollectionViewLayout(createLayout(), animated: false)
+    }
+
+    override func viewLayoutMarginsDidChange() {
+        super.viewLayoutMarginsDidChange()
+
+        collectionView.setCollectionViewLayout(createLayout(), animated: false)
+    }
```
Comment (Clstroud) [inline Vocable/Features/Settings/TimingSensitivity/TimingSensitivityViewController.swift]: I _believe_ using the section provider version of compositional layout would give this to you for free, like in SettingsViewController. That way the layout can invalidate itself and you'd just make sure your sections are properly configured based on whatever the environment is at the time.

File: Vocable/Features/Settings/TimingSensitivity/TimingSensitivityViewController.swift [hunk 2]
```diff
@@ -8,12 +8,160 @@
...
+
+    func collectionView(_ collectionView: UICollectionView, shouldHighlightItemAt indexPath: IndexPath) -> Bool {
+        guard let item = dataSource.itemIdentifier(for: indexPath) else { return false }
+        switch item {
+        case .dwellTime, .sensitivity:
+            return false
+        }
```
Comment (Clstroud) [inline Vocable/Features/Settings/TimingSensitivity/TimingSensitivityViewController.swift:99]:
```suggestion
        return false
```

File: Vocable/Features/Settings/TimingSensitivity/TimingSensitivityViewController.swift [hunk 3]
```diff
@@ -8,12 +8,160 @@
...
+    }
+
+    func collectionView(_ collectionView: UICollectionView, shouldSelectItemAt indexPath: IndexPath) -> Bool {
+        guard let item = dataSource.itemIdentifier(for: indexPath) else { return false }
+        switch item {
+        case .dwellTime, .sensitivity:
+            return false
+        }
```
Comment (Clstroud) [inline Vocable/Features/Settings/TimingSensitivity/TimingSensitivityViewController.swift:107]:
```suggestion
        return false
```

File: Vocable/Features/Settings/TimingSensitivity/TimingSensitivityViewController.swift [hunk 4]
```diff
@@ -8,12 +8,160 @@
...
+            cell.decreaseTimeButton.addTarget(self,
+                                              action: #selector(self.handleDecreasingDwellTime(_:)),
+                                              for: .primaryActionTriggered)
+            cell.increaseTimeButton.addTarget(self,
+                                              action: #selector(self.handleIncreasingDwellTime(_:)),
+                                              for: .primaryActionTriggered)
+            return cell
+        case .sensitivity:
+            let cell = collectionView.dequeueReusableCell(withReuseIdentifier: SensitivityCollectionViewCell.reuseIdentifier, for: indexPath) as! SensitivityCollectionViewCell
+            if traitCollection.horizontalSizeClass == .compact && traitCollection.verticalSizeClass == .regular {
```
Comment (Clstroud) [inline Vocable/Features/Settings/TimingSensitivity/TimingSensitivityViewController.swift:123]:
```suggestion
            if sizeClass == hCompact_vRegular {
```

**Review by Clstroud — APPROVED** [review]
Some minor suggestions, some of which (layout-related stuff) could be in a followup PR

**Comment by stechkev** [comment]
Back "<-" button not lined up properly

**Pull request merged**